src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -218,21 +186,188 @@ impl WfContext {
     }
 }
 
-struct WFCommandFut {
+/// A Future that can be cancelled.
+/// Used in the prototype SDK for cancelling operations like timers and activities.
+pub trait CancellableFuture<T>: Future<Output = T> {
+    /// Cancel this Future
+    fn cancel(&self, cx: &WfContext);
+}
+
+struct WFCommandFut<T>
+where
+    T: From<UnblockEvent>,
+{
+    /// We need a field that references the template type T, this works
+    unused: PhantomData<T>,
+    cancellable_id: CancellableID,
     result_rx: oneshot::Receiver<UnblockEvent>,
 }
 
-impl WFCommandFut {
-    fn new() -> (Self, oneshot::Sender<UnblockEvent>) {
+impl<T> WFCommandFut<T>
+where
+    T: From<UnblockEvent>,
+{
+    fn new(cancellable_id: CancellableID) -> (Self, oneshot::Sender<UnblockEvent>) {
         let (tx, rx) = oneshot::channel();
-        (Self { result_rx: rx }, tx)
+        (
+            Self {
+                cancellable_id,
+                result_rx: rx,
+                unused: PhantomData,
+            },
+            tx,
+        )
     }
 }
 
-impl Future for WFCommandFut {
-    type Output = UnblockEvent;
+impl<T> Unpin for WFCommandFut<T> where T: From<UnblockEvent> {}
```

Comment:
I believe this should not be necessary, it should get auto-derived

Reply:
It became necessary, complier was complaining, I think it's the phantom data.